### PR TITLE
Add replace image button to Collage block gallery items

### PR DESCRIPTION
**Pull request by jrtashjian**
### Description
Adds the ability to replace an image with a button in the Collage gallery.

### Screenshots
image

### Types of changes
- New feature

### How has this been tested?
Manually and E2E tests

### Checklist:
- [x] My code is tested
- [x] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [x] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: src/blocks/gallery-collage/styles/style.scss [hunk 1]
```diff
@@ -1,3 +1,4 @@
+.editor-styles-wrapper .wp-block-coblocks-gallery-collage,
```
Comment (AnthonyLedesma) [inline src/blocks/gallery-collage/styles/style.scss:1]: I am not certain the purpose of this change. Shouldn't the `.wp-block-coblocks-gallery-collage {` selector successfully apply to both editor and front-end blocks?

**Review by AnthonyLedesma — COMMENTED** [review]
Changes look good to merge. Just one question on here regarding a CSS selector change.

File: src/blocks/gallery-collage/styles/style.scss (hunk 1 above)
Comment (jrtashjian) [reply src/blocks/gallery-collage/styles/style.scss:1]: The only purpose is to increase CSS specificity to avoid using `!important`

**Review by AnthonyLedesma — CHANGES_REQUESTED** [review]
How do you feel about addressing this ESLint warning? 
https://app.circleci.com/pipelines/github/godaddy-wordpress/coblocks/5643/workflows/dbf1be8c-c87e-4131-9fc5-d3b69ddaca10/jobs/21319

I tried playing with rules some but wasn't able to achieve the behavior I was looking for. I ended up leaving comments instead.
https://github.com/godaddy-wordpress/coblocks/blob/2881998f52ce797833f867474abe313bfefd06dd/src/blocks/author/test/author.cypress.js#L73-L74

**Review by AnthonyLedesma — APPROVED** [review]

**Pull request merged**